### PR TITLE
[fix] Out-of-bounds reads

### PR DESCRIPTION
**Pull request by ace20022**

### CODE REVIEW

**Comment by ace20022** [comment]
@FernetMenta I'm not sure about https://github.com/ace20022/xbmc/commit/31cf340ef6959b25c86eefea8f0281e4c665633a . Should I ask upstream instead?

File: xbmc/cores/dvdplayer/DVDCodecs/Overlay/contrib/cc_decoder708.cpp [hunk 1]
```diff
@@ -290,6 +290,7 @@ void updateScreen (cc708_service_decoder *decoder)
       left=0;
     int copyrows=top + wnd[i]->row_count >= I708_SCREENGRID_ROWS ?
         I708_SCREENGRID_ROWS - top : wnd[i]->row_count;
+    copyrows = copyrows > I708_MAX_ROWS ? I708_MAX_ROWS : copyrows;
```
Comment (arnova) [inline xbmc/cores/dvdplayer/DVDCodecs/Overlay/contrib/cc_decoder708.cpp]: Might as well make that "if copyrows > I708_MAX_ROWS copyrows = I708_MAX_ROWS " , or use std::min

**Comment by ace20022** [comment]
> Are you able to construct a case where this code fails? ... all parameter we use are already forced inside bounds.

That's true, but changing the init code may break it (albeit unlikely here) and hardly anyone checks all places where the array is accessed afterwards.
With that attitude we can abstain from using coverity or the like.

**Comment by FernetMenta** [comment]
> With that attitude we can abstain from using coverity or the like.

What attitude? Stop bitching at me. Am I not allowed to express my concerns?

**Comment by FernetMenta** [comment]
@ace20022 

> Are you able to construct a case where this code fails?

That was only a question. I only was interested in whether the current code would allow out-of-bounds, nothing more. No judgement of this PR. The code you changed is not our code and I think in respect to the original author we should place a comment in the code that this is a result from using a tool like coverity and not a fix for a real scenario of what the commit message states.

**Comment by ace20022** [comment]
> Stop bitching at me.

Please mind your words.

> Am I not allowed to express my concerns?

Of course you are. 

> ... and not a fix for a real scenario of what the commit message states.

You are right, the commit message is imprecise, it is missing the word "theoretical".

**Pull request closed without merging**